dri3_ws.c

```diff
@@ -399,22 +399,17 @@ static void handle_special_event(struct driws_drawable *drawable, xcb_present_ge
 
 		struct driws_buffer *buffer = find_buffer_from_list(ie->pixmap);
 
-		FAIL_IF(!buffer, "no buffer");
-
-		buffer->busy = false;
+		if (buffer)
+			buffer->busy = false;
```

Comment:
Does this commit change the behavior so that we can have buffer == NULL here? Or is the above change separate from this commit?

Reply:
Yes, this now seems to be a possibility, as the buffer is being destroyed